src/Okta.Sdk/Internal/UserAgentBuilder.cs

```diff
@@ -83,6 +83,31 @@ private string Generate()
             .Trim();
         }
 
+        private static string GetFrameworkDescription()
+        {
+            var frameworkDescription = RuntimeInformation.FrameworkDescription;
+
+            // RuntimeInformation.FrameworkDescription only returns the CLI version for .NET Core.
+            // We need this workaround to get the product version and have a useful report.
+            if (RuntimeInformation.FrameworkDescription.StartsWith(".NET Core", StringComparison.OrdinalIgnoreCase))
+            {
+                var runtimeAssembly = typeof(System.Runtime.GCSettings).GetTypeInfo().Assembly;
+
+                if (runtimeAssembly != null)
+                {
+                    var runtimeAssemblyLocation = runtimeAssembly.CodeBase.Split(new[] { '/', '\\' }, StringSplitOptions.RemoveEmptyEntries);
+                    int dotnetCoreAppIndex = Array.IndexOf(runtimeAssemblyLocation, "Microsoft.NETCore.App");
+
+                    if (dotnetCoreAppIndex > 0 && dotnetCoreAppIndex < runtimeAssemblyLocation.Length - 2)
+                    {
+                        frameworkDescription = $".NET Core {runtimeAssemblyLocation[dotnetCoreAppIndex + 1]}";
+                    }
+                }
+            }
+
+            return frameworkDescription;
+        }
```

Comment:
I would recommend adding some unit tests for the string manipulation logic.

Usually I don't spend a lot of time asking for unit tests in code reviews because in many cases they don't really test much. But string manipulation is one of the few examples where unit tests are actually really useful. It's easy to have a bug and also easy to test.